### PR TITLE
Simbrief Edit & Download Latest OFP

### LINKED ISSUE #1226: Download Latest/Edited SimBrief OFP Pack with API
**Is your feature request related to a problem? Please describe.**
No, it is a new feature request.

**Describe the feature you'd like**
As you may know, it is possible to edit a SimBrief OFP once it is generated and API (according to its docs) allows re-downloading an edited ofp pack.

Is it possible to add this feature to our own API integration so we would be able to edit a generated ofp (via SimBrief interface of course) and then with a click of a button we can download that latest ofp to our database (with updating current record)

We can build a `static_id` as per SimBrief requirements easily for generated flights with our current form like 
`$user->id."_".$flight->id` or (if it needs a shorter string) `$user->id."_".$flight->number` 
then we can use that static_id we created to pull the latest/edited/regenerated ofp.  

**Describe alternatives you've considered**
None, continue using current code, edit if needed and download manually and use that downloaded file with vmsacars.

### PR DESCRIPTION
**Pull request by FatihKoz**
Changes aim to have the ability of editing generated SimBrief Flight Plans and re-downloading the latest one.

Closes #1226 

sb_default

### CODE REVIEW

File: app/Services/SimBriefService.php [hunk 1]
```diff
@@ -39,10 +41,18 @@ public function downloadOfp(
...
+        array $fares = [],
+        string $sb_userid = null,
+        string $sb_static_id = null
     ) {
         $uri = str_replace('{id}', $ofp_id, config('phpvms.simbrief_url'));
 
+        if ($sb_userid && $sb_static_id) {
+            // $uri = str_replace('{sb_userid}', $sb_userid, config('phpvms.simbrief_update_url'));
+            // $uri = str_replace('{sb_static_id}', $sb_static_id, $uri);
+            $uri = 'https://www.simbrief.com/api/xml.fetcher.php?userid='.$sb_userid.'&static_id='.$sb_static_id;
```
Comment (nabeelio) [inline app/Services/SimBriefService.php:53]: Use the URL that's on the config file; you've got the commented out code up there

File: config/phpvms.php [hunk 2]
```diff
@@ -52,6 +52,11 @@
      */
     'simbrief_url' => 'https://www.simbrief.com/ofp/flightplans/xml/{id}.xml',
 
+    /*
+     * URL for fetching an updated Simbrief flight plan via API
+     */
+    'simbrief_update_url' => 'https://www.simbrief.com/api/xml.fetcher.php?userid={sb_userid}&static_id={sb_static_id}',
```
Comment (nabeelio) [inline config/phpvms.php:58]: `sb_user_id`

File: resources/views/layouts/default/flights/simbrief_briefing.blade.php [hunk 3]
```diff
@@ -284,6 +292,31 @@ class="btn btn-outline-primary pull-right"
...
+  @if(!empty($simbrief->xml->params->static_id))
+    <div class="modal fade" id="OFP_Edit" data-backdrop="static" data-keyboard="false" tabindex="-1" aria-labelledby="staticBackdropLabel" aria-hidden="true">
+      <div class="modal-dialog" style="max-width: 1020px;">
+        <div class="modal-content p-0" style="border-radius: 5px;">
+          <div class="modal-header p-1">
+            <h5 class="modal-title m-1 p-0">SimBrief</h5>
+            <span class="close"><i class="fas fa-times-circle" data-dismiss="modal" aria-label="Close" aria-hidden="true"></i></span>
+          </div>
+          <div class="modal-body p-0">
+            <iframe src="https://www.simbrief.com/system/dispatch.php?editflight=last&static_id={{ $simbrief->xml->params->static_id }}" style="width: 100%; height: 80vh;" frameBorder="0" title="SimBrief"></iframe>
```
Comment (nabeelio) [inline resources/views/layouts/default/flights/simbrief_briefing.blade.php:306]: Is the iframe the right way to do this? Just makes sure against their API docs - I seem to remember something about the iframe being discouraged. Also, make the simbrief URL a config option and have the full URL be passed in via controller

File: resources/views/layouts/default/flights/simbrief_form.blade.php [hunk 4]
```diff
@@ -160,6 +160,7 @@
               <input type="hidden" name="omit_stars" value="0">
               <input type="hidden" name="cruise" value="CI">
               <input type="hidden" name="civalue" value="AUTO">
+              <input type="hidden" name="static_id" value="{{ $user->ident.'_'.$flight->id }}">
```
Comment (nabeelio) [inline resources/views/layouts/default/flights/simbrief_form.blade.php:163]: Have this static ID generated in the controller. It's easier to find.

File: app/Services/SimBriefService.php (hunk 1 above)
Comment (FatihKoz) [reply app/Services/SimBriefService.php:53]: That was the problem i faced, when I use the config (commented out lines) it generates an error which I was not able to solve. And this was the reason I wrote a note about this line in the main issue text, kindly requested you to check that part.

File: resources/views/layouts/default/flights/simbrief_briefing.blade.php (hunk 3 above)
Comment (FatihKoz) [reply resources/views/layouts/default/flights/simbrief_briefing.blade.php:306]: There is not other way to edit a generated flight plan, and this url is directly mentioned in the API docs for editing them.

File: resources/views/layouts/default/flights/simbrief_form.blade.php (hunk 4 above)
Comment (FatihKoz) [reply resources/views/layouts/default/flights/simbrief_form.blade.php]: Done

File: config/phpvms.php (hunk 2 above)
Comment (FatihKoz) [reply config/phpvms.php]: changed

File: app/Services/SimBriefService.php (hunk 1 above)
Comment (FatihKoz) [reply app/Services/SimBriefService.php]: `[2021-06-06 13:18:48] prod.ERROR: simplexml_load_string(): Entity: line 1: parser error : Start tag expected, '&lt;' not found
{"userId":3,"exception":"[object] (ErrorException(code: 0): simplexml_load_string(): Entity: line 1: parser error : Start tag expected,
'&lt;' not found at /app/Services/SimBriefService.php:73)`

This is the error when switched to config for reading the url

Comment (nabeelio) [reply app/Services/SimBriefService.php]: Did you output the URL after the str_replace to make sure the replace was completed properly?

**Pull request merged**